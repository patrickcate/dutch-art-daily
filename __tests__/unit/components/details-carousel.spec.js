import DetailsCarousel from '@components/details-carousel.vue'

describe('DetailsCarousel Component', () => {
  let wrapper

  beforeEach(() => {
    wrapper = createWrapper(
      DetailsCarousel,
      {},
      {
        state: {
          currentDetailsHeight: null,
        },
      }
    )
  })

  it('should be a Vue instance', () => {
    expect(wrapper.exists()).toBeTruthy()
  })

  it('renders correctly', async () => {
    wrapper.vm.$store.replaceState({
      ...wrapper.vm.$store.state,
      currentDetailsHeight: 100,
    })
    await wrapper.vm.$nextTick()

    expect(wrapper).toMatchSnapshot()
  })

  it('updates height from store state', async () => {
    wrapper.vm.$store.replaceState({
      ...wrapper.vm.$store.state,
      currentDetailsHeight: 100,
    })
    await wrapper.vm.$nextTick()

    expect(wrapper.attributes('style')).toContain('height: 100px;')

    wrapper.vm.$store.state.currentDetailsHeight = 200
    await wrapper.vm.$nextTick()

    expect(wrapper.attributes('style')).toContain('height: 200px;')
  })
})
