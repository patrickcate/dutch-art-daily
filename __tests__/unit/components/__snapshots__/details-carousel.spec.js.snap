// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailsCarousel Component renders correctly 1`] = `
<base-carousel-stub name="details" style="height: 100px;">
  <details-slide-stub id="12-26"></details-slide-stub>
  <details-slide-stub id="12-27"></details-slide-stub>
  <details-slide-stub id="12-28"></details-slide-stub>
  <details-slide-stub id="12-29"></details-slide-stub>
  <details-slide-stub id="12-30"></details-slide-stub>
  <details-slide-stub id="12-31"></details-slide-stub>
  <details-slide-stub id="01-01"></details-slide-stub>
</base-carousel-stub>
`;
